docs/snippets/chains/API-reference.mdx

```diff
@@ -0,0 +1,572 @@
+# API Reference
+
+## *class* `truss_chains.Assets`
+
+Specifies which assets a chainlet can access in the remote deployment.
+
+Model weight caching can be used like this:
+
+```python
+import truss_chains as chains
+from truss import truss_config
+
+mistral_cache = truss_config.ModelRepo(
+  repo_id="mistralai/Mistral-7B-Instruct-v0.2",
+  allow_patterns=["*.json", "*.safetensors", ".model"]
+  )
+chains.Assets(cached=[mistral_cache], ...)
+```
+
+See [truss caching guide](https://truss.baseten.co/guides/model-cache#enabling-caching-for-a-model)
+for more details on caching.
+
+
+**Parameters:**
+
+| Name | Type | Description |
+|------|------|-------------|
+| `cached` | *Iterable[ModelRepo]* | One or more `truss_config.ModelRepo` objects. |
+| `secret_keys` | *Iterable[str]* | Names of secrets stored on baseten, that the chainlet should have access to. You can manage secrets on baseten [here](https://app.baseten.co/settings/secrets). |
+
+
+#### get_spec()
+
+Returns parsed and validated assets.
+
+* **Return type:**
+  *AssetSpec*
+
+## *class* `truss_chains.ChainletBase`
+
+Base class for all chainlets.
+
+Inheriting from this class adds validations to make sure subclasses adhere to the
+chainlet pattern and facilitates remote chainlet deployment.
+
+Refer to [the docs](https://truss.baseten.co/chains/getting-started) and this
+[example chainlet](https://github.com/basetenlabs/truss/blob/main/truss-chains/truss_chains/example_chainlet.py)
+for more guidance on how to create subclasses.
+
+## *class* `truss_chains.deploy.ChainService`
```

Comment:
should this be exposed in the docs? I feel like `add_service` and whatnot are all somewhat internal implementation details of the CLI, I don't know if we want to codify/expose these to the outside world in their current form at least.

Reply:
Yeah, I thought about this too.
The reason why `ChainService` is at all part of the docs is because it's the return value of the (public API) function `deploy_remotely`. But some of its methods must/should not be used by users.

It seems to me that this is actually a bit a leaky design and we should ideally have something like an **internal** `ChjainServiceBuilder` (with methods like `add_service`) and then convert that to a **public** `ChainService` which only has the methods relevant to users like `predict` - it would be quite a bit of refactoring work which I don't wanna prioritize right now.

We could as a patchy solution just un-document the methods not intended for public use or keep them, but add in the docstring that they are not intended for end-users.

What do you think is overall best?